client/src/commons/AlphanumericTextField/AlphanumericTextField.tsx

```diff
@@ -14,52 +12,12 @@ const stringAlphanum = yup
 const errorMessage = 'השדה יכול להכיל רק אותיות ומספרים';
 
 const AlphanumericTextField: AlphanumericTextFieldType = (props) => {
-  const value = (props.value == null || props.value === undefined) ? "" : props.value;
-  const {
-    name,
-    onChange,
-    onBlur,
-    setError,
-    clearErrors,
-    errors,
-    placeholder,
-    className,
-    label,
-    required,
-    testId,
-  } = props;
-
-  const conditionalyTriggerOnChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const newValue = e.target.value;
-    const isValid = stringAlphanum.isValidSync(newValue);
-    if (isValid || newValue === "") {
-      clearErrors(name);
-      onChange(newValue);
-    } else {
-      setError(name, {
-        type: "manual",
-        message: errorMessage,
-      });
-    }
-  };
-
-  const errorObject = get(errors, name);
-
   return (
-    <Tooltip open={errorObject? true : false} title={errorObject ? errorMessage : ""}>
-      <TextField
-        test-id={testId}
-        required={required}
-        name={name}
-        value={value}
-        onChange={conditionalyTriggerOnChange}
-        onBlur={onBlur}
-        error={errorObject ? true : false}
-        label={errorObject ? errorObject.message : label}
-        placeholder={placeholder}
-        className={className}
-      />
-    </Tooltip>
+    <TypePreventiveTextField
```

Comment:
can we delete AlphanumericTextField file? (changed to AlphabetTextFieldType)

Reply:
There are two types: text field which only allows letters and a text field that allows letters and numbers.

Reply:
V

Reply:
so can we use TypePreventiveTextField ?

Reply:
TypePreventiveTextField is a generic component which accepts a `yup` schema and prevents typing according to it.
It was created so AlphanumericTextField and AlphabetTextField can be built from it. If there is a future need for more types of typing prevention we can now create components from TypePreventiveTextField to achieve that.